lib/net.c

```diff
@@ -154,120 +154,196 @@ ethertype_filter_add(struct gatekeeper_if *iface, uint16_t ether_type,
 		.flags = RTE_5TUPLE_FLAGS,
-		.dst_ip = dst_ip,
+		.dst_ip = dst_ip_be,
 		.dst_ip_mask = UINT32_MAX,
 		.src_ip = 0,
 		.src_ip_mask = 0,
-		.dst_port = dst_port,
-		.dst_port_mask = dst_port_mask,
-		.src_port = src_port,
-		.src_port_mask = src_port_mask,
-		.proto = proto,
-		.proto_mask = UINT8_MAX,
-		.tcp_flags = 0,
-		.priority = 1,
-		.queue = queue_id,
-	};
-
-	struct rte_eth_ntuple_filter filter_v6 = {
-		.flags = RTE_5TUPLE_FLAGS,
-		.dst_ip = 0,
-		.dst_ip_mask = 0,
-		.src_ip = 0,
-		.src_ip_mask = 0,
-		.dst_port = dst_port,
-		.dst_port_mask = dst_port_mask,
-		.src_port = src_port,
-		.src_port_mask = src_port_mask,
+		.dst_port = dst_port_be,
+		.dst_port_mask = dst_port_mask_be,
+		.src_port = src_port_be,
+		.src_port_mask = src_port_mask_be,
 		.proto = proto,
 		.proto_mask = UINT8_MAX,
 		.tcp_flags = 0,
 		.priority = 1,
 		.queue = queue_id,
 	};
 
-	RTE_VERIFY(iface->hw_filter_ntuple);
-
-	if (!ipv4_configured)
-		goto ipv6;
-
 	for (i = 0; i < iface->num_ports; i++) {
 		uint16_t port_id = iface->ports[i];
-		ret = rte_eth_dev_filter_ctrl(port_id,
+		int ret = rte_eth_dev_filter_ctrl(port_id,
 			RTE_ETH_FILTER_NTUPLE,
 			RTE_ETH_FILTER_ADD,
 			&filter_v4);
 		if (ret == -ENOTSUP) {
 			G_LOG(ERR,
 				"net: hardware doesn't support adding an IPv4 ntuple filter on port %hhu\n",
 				port_id);
-			ret = -1;
-			goto out;
+			return -1;
 		} else if (ret == -ENODEV) {
 			G_LOG(ERR,
 				"net: port %hhu is invalid for adding an IPv4 ntuple filter\n",
 				port_id);
-			ret = -1;
-			goto out;
+			return -1;
 		} else if (ret != 0) {
 			G_LOG(ERR,
 				"net: other errors that depend on the specific operations implementation on port %hhu for adding an IPv4 ntuple filter\n",
 				port_id);
-			ret = -1;
-			goto out;
+			return -1;
 		}
 	}
-ipv6:
-	if (!ipv6_configured)
-		goto out;
 
-	for (i = 0; i < iface->num_ports; i++) {
-		uint16_t port_id = iface->ports[i];
-		ret = rte_eth_dev_filter_ctrl(port_id,
-			RTE_ETH_FILTER_NTUPLE,
-			RTE_ETH_FILTER_ADD,
-			&filter_v6);
-		if (ret == -ENOTSUP) {
-			G_LOG(ERR,
-				"net: hardware doesn't support adding an IPv6 ntuple filter on port %hhu\n",
-				port_id);
-			ret = -1;
-			goto out;
-		} else if (ret == -ENODEV) {
-			G_LOG(ERR,
-				"net: port %hhu is invalid for adding an IPv6 ntuple filter\n",
-				port_id);
-			ret = -1;
-			goto out;
-		} else if (ret != 0) {
-			G_LOG(ERR,
-				"net: other errors that depend on the specific operations implementation on port %hhu for adding an IPv6 ntuple filter\n",
-				port_id);
-			ret = -1;
-			goto out;
+	return 0;
+}
+
+/*
+ * @dst_ip_be, @src_port_be, @src_port_mask_be, @dst_port_be, and
+ * @dst_port_mask_be must be in big endian -- this function
+ * converts the values to host ordering as needed.
+ */
+static void
+ipv4_fill_acl_rule(struct ipv4_acl_rule *rule,
+	rte_be32_t dst_ip_be,
+	rte_be16_t src_port_be, rte_be16_t src_port_mask_be,
+	rte_be16_t dst_port_be, rte_be16_t dst_port_mask_be,
```

Comment:
We can drop the code comment above. The same explanation presented in the previous review comment.